scripts/gitlab/build_and_test.sh

```diff
@@ -152,3 +156,10 @@ then
         echo "ERROR: failure(s) while running CTest" && exit 1
     fi
 fi
+
+echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+echo "~~~~~ CLEAN UP"
+echo "~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~"
+
+make clean
+rm -rf uberenv_libs
```

Comment:
The drawback of cleaning here is that the clean won't apply to any failing job.
You could put the clean commands in the "after_script" section of the CI job insteald.